src/timestamp.rs

```diff
@@ -164,148 +164,187 @@ impl Interval {
-    }
+        #[test]
+        fn test_ns_spaces() {
+            assert_eq!(Timestamp::parse("  567.0  ns  "), Ok(Timestamp(567)));
+        }
 
-    #[test]
-    fn test_no_value() {
-        assert_eq!(
-            Timestamp::parse("ms"),
-            Err(TimestampParseError::InvalidValue)
-        );
-    }
+        #[test]
+        fn test_no_unit() {
+            assert_eq!(Timestamp::parse("500.0"), Err(TimestampParseError::NoUnit));
+        }
 
-    #[test]
-    fn test_invalid_unit() {
-        assert_eq!(
-            Timestamp::parse("500.0 foo"),
-            Err(TimestampParseError::InvalidUnit)
-        );
-    }
+        #[test]
+        fn test_no_value() {
+            assert_eq!(
+                Timestamp::parse("ms"),
+                Err(TimestampParseError::InvalidValue)
+            );
+        }
 
-    #[test]
-    fn test_invalid_value() {
-        assert_eq!(
-            Timestamp::parse("foo ms"),
-            Err(TimestampParseError::InvalidValue)
-        );
-    }
+        #[test]
+        fn test_invalid_unit() {
+            assert_eq!(
+                Timestamp::parse("500.0 foo"),
+                Err(TimestampParseError::InvalidUnit)
+            );
+        }
 
-    #[test]
-    fn test_invalid_value2() {
-        assert_eq!(
-            Timestamp::parse("500.0.0 ms"),
-            Err(TimestampParseError::InvalidValue)
-        );
-    }
+        #[test]
+        fn test_invalid_value() {
+            assert_eq!(
+                Timestamp::parse("foo ms"),
+                Err(TimestampParseError::InvalidValue)
+            );
+        }
+
+        #[test]
+        fn test_invalid_value2() {
+            assert_eq!(
+                Timestamp::parse("500.0.0 ms"),
+                Err(TimestampParseError::InvalidValue)
+            );
+        }
 
-    #[test]
-    fn test_invalid_value3() {
-        assert_eq!(
-            Timestamp::parse("500.0.0"),
-            Err(TimestampParseError::NoUnit)
-        );
+        #[test]
+        fn test_invalid_value3() {
+            assert_eq!(
+                Timestamp::parse("500.0.0"),
+                Err(TimestampParseError::NoUnit)
+            );
+        }
+
+        #[test]
+        fn test_extra() {
+            assert_eq!(
+                Timestamp::parse("500.0 ms asdfadf"),
+                Err(TimestampParseError::InvalidUnit)
+            );
+        }
     }
 
-    #[test]
-    fn test_extra() {
-        assert_eq!(
-            Timestamp::parse("500.0 ms asdfadf"),
-            Err(TimestampParseError::InvalidUnit)
-        );
+    mod interval {
+        use super::*;
+
+        #[test]
+        fn test_translate_positive() {
+            let start: Timestamp = Timestamp::parse("234.5 ms").unwrap();
+            let end: Timestamp = Timestamp::parse("235.5 ms").unwrap();
+            let expected_start: Timestamp = Timestamp(start.0 + 250);
+            let expected_end: Timestamp = Timestamp(end.0 + 250);
```

Comment:
Same with removing types here. These definitely can be inferred, so we don't need them.

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/27/commits/5c0d57a8d4d463a3185ac7c598084d00fa16c729